src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java

```diff
@@ -37,17 +69,282 @@ public WorkOrderBuildDecoration(final String structureName, final String workOrd
+     */
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        buildingLocation = BlockPosUtil.readFromNBT(compound, TAG_BUILDING);
+        final Structures.StructureName sn = new Structures.StructureName(compound.getString(TAG_SCHEMATIC_NAME));
+        structureName = sn.toString();
+        workOrderName = compound.getString(TAG_WORKORDER_NAME);
+        cleared = compound.getBoolean(TAG_IS_CLEARED);
+        md5 = compound.getString(TAG_SCHEMATIC_MD5);
+        if (!Structures.hasMD5(structureName))
+        {
+            // If the schematic move we can use the MD5 hash to find it
+            final Structures.StructureName newSN = Structures.getStructureNameByMD5(md5);
+            if (newSN == null)
+            {
+                Log.getLogger().error("WorkOrderBuildDecoration.readFromNBT: Could not find " + structureName);
+            }
+            else
+            {
+                Log.getLogger().warn("WorkOrderBuildDecoration.readFromNBT: replace " + sn + " by " + newSN);
+                structureName = newSN.toString();
+
+            }
+        }
+
+        buildingRotation = compound.getInteger(TAG_BUILDING_ROTATION);
+        requested = compound.getBoolean(TAG_IS_REQUESTED);
+        isMirrored = compound.getBoolean(TAG_IS_MIRRORED);
     }
 
+    /**
+     * Save the Work Order to an NBTTagCompound.
+     *
+     * @param compound NBT tag compound.
+     */
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.writeToNBT(compound);
+        BlockPosUtil.writeToNBT(compound, TAG_BUILDING, buildingLocation);
+        if (workOrderName != null)
+        {
+            compound.setString(TAG_WORKORDER_NAME, workOrderName);
+        }
+        compound.setBoolean(TAG_IS_CLEARED, cleared);
+        if (md5 != null)
+        {
+            compound.setString(TAG_SCHEMATIC_MD5, md5);
+        }
+        if (structureName == null)
+        {
+            Log.getLogger().error("WorkOrderBuild.writeToNBT: structureName should not be null!!!");
+        }
+        else
+        {
+            compound.setString(TAG_SCHEMATIC_NAME, structureName);
+        }
+        compound.setInteger(TAG_BUILDING_ROTATION, buildingRotation);
+        compound.setBoolean(TAG_IS_REQUESTED, requested);
+        compound.setBoolean(TAG_IS_MIRRORED, isMirrored);
+    }
+
+    /**
+     * Attempt to fulfill the Work Order.
+     * Override this with an implementation for the Work Order to find a Citizen to perform the job
+     * <p>
+     * finds the first suitable builder for this job.
+     *
+     * @param colony The colony that owns the Work Order.
+     */
+    @Override
+    public void attemptToFulfill(@NotNull final Colony colony)
+    {
+        boolean sendMessage = true;
+        boolean hasBuilder = false;
+
+        for (@NotNull final CitizenData citizen : colony.getCitizens().values())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)